testing/integration/otel_test.go

```diff
@@ -1595,3 +1597,323 @@ service:
+    verbosity: detailed
+  elasticsearch/log:
+    endpoints:
+      - {{.ESEndpoint}}
+    compression: none
+    api_key: {{.ESApiKey}}
+    logs_dynamic_index:
+      enabled: true
+    batcher:
+      enabled: true
+      flush_timeout: 1s
+    mapping:
+      mode: bodymap	  
+service:
+  pipelines:
+    logs:
+      receivers:
+        - filebeatreceiver/filestream-monitoring-agent
+      exporters:
+        - elasticsearch/log
+`
+
+	beatsApiKey, err := base64.StdEncoding.DecodeString(esApiKey.Encoded)
+	require.NoError(t, err, "error decoding api key")
+
+	var configBuffer bytes.Buffer
+	require.NoError(t,
+		template.Must(template.New("config").Parse(configTemplate)).Execute(&configBuffer,
+			configOptions{
+				InputPath:      inputPath,
+				HomeDir:        tmpDir,
+				ESEndpoint:     esEndpoint,
+				ESApiKey:       esApiKey.Encoded,
+				BeatsESApiKey:  string(beatsApiKey),
+				SocketEndpoint: socketEndpoint,
+			}))
+	configContents := configBuffer.Bytes()
+	t.Cleanup(func() {
+		if t.Failed() {
+			t.Logf("Contents of agent config file:\n%s\n", string(configContents))
+		}
+	})
+
+	err = fixture.Configure(ctx, configContents)
+	require.NoError(t, err)
+
+	cmd, err := fixture.PrepareAgentCommand(ctx, nil)
+	require.NoError(t, err)
+	cmd.WaitDelay = 1 * time.Second
+
+	var output strings.Builder
+	cmd.Stderr = &output
+	cmd.Stdout = &output
+
+	err = cmd.Start()
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		if t.Failed() {
+			t.Log("Elastic-Agent output:")
+			t.Log(output.String())
+		}
+	})
+
+	require.Eventually(t, func() bool {
+		err = fixture.IsHealthy(ctx)
+		if err != nil {
+			t.Logf("waiting for agent healthy: %s", err.Error())
+			return false
+		}
+		return true
+	}, 30*time.Second, 1*time.Second)
+
+	// check if elastic-agent log path exists
+	// This is important for the test to succeed
+	pattern := inputPath + "/data/elastic-agent-*/logs/elastic-agent-*.ndjson"
+	files, err := filepath.Glob(pattern)
+	if err != nil {
+		t.Fatalf("Error checking files: %v", err)
+	}
+
+	if len(files) == 0 {
+		t.Errorf("No matching files found in path: %s", pattern)
+	}
+
+	var agentDocs estools.Documents
+	var otelDocs estools.Documents
+	require.Eventually(t,
+		func() bool {
+			findCtx, findCancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer findCancel()
+
+			agentDocs, err = estools.GetLogsForIndexWithContext(findCtx, info.ESClient, ".ds-"+fbMonitoringIndex+"*", map[string]interface{}{
+				"data_stream.dataset": "elastic_agent",
+			})
+
+			require.NoError(t, err)
+			otelDocs, err = estools.GetLogsForIndexWithContext(findCtx, info.ESClient, ".ds-"+fbReceiverMonitoringIndex+"*", map[string]interface{}{
+				"data_stream.dataset": "otel",
+			})
+			require.NoError(t, err)
+
+			// Atleast 1 Document in each index should be present
+			return otelDocs.Hits.Total.Value > 1 && agentDocs.Hits.Total.Value > 1
+		},
+		2*time.Minute, 1*time.Second, "the number of logs in both index are not same")
```

Comment:
```suggestion
		2*time.Minute, 1*time.Second, "a datastream is empty")
```
We don't validate counts are equal